### ISSUE TITLE
Need Index data

### ISSUE THREAD

**Issue by LRagji** [girishg4t/bhavCopy-downloader#28]
I think nifty 50 historical index data is missing ,5min etc
Also the git which hosts the data cannot be cloned it says  `fatal: BUG: initial ref transaction called with existing refs`

**Comment by girishg4t** [comment]
Thanks Laukik for the issue
1) We are downloading only EOD data for NSE and BSE, however we can also include 5min etc data if you want, could you please give us the link from NSE to show which data you want
2) It is public repo so any one can clone that, if you facing any issue give me the steps so that i can try to reproduce.
only thing is it is a large repo so downloading takes time

**Comment by LRagji** [comment]
@girishg4t  
1. EOD data is fine i am doing backtesting so not an issue, for the source of data i got it from [kaggle](https://www.kaggle.com/datasets/debashis74017/nifty-50-minute-data) not sure how to get that on NSE.
2. For the repo error i just did
`git clone https://github.com/girishg4t/nse-bse-bhavcopy.git` and got the error stated above..

**Comment by LRagji** [comment]
@girishg4t  can you get options data from here https://www.nseindia.com/get-quotes/derivatives?symbol=BANKNIFTY ?

**Comment by girishg4t** [comment]
Yes there is this api though which we can download that data, Please let me know this is  what you want?
```

curl --location --request GET 'https://www.nseindia.com/api/option-chain-indices?symbol=BANKNIFTY' \
--header 'Cookie: ak_bmsc=E26E3B324A2A6745BFCDFABC
[banknifty.json.zip](https://github.com/girishg4t/bhavCopy-downloader/files/12139004/banknifty.json.zip)
CB261ABC~000000000000000000000000000000~YAAQlidzaFuNBYGJAQAAp4lIgxRNM3RTnt3e9zUIWHJ/qHkvpG19CgRjVB7Tb7eaIfpHMJe4P/3f/efD1SHk/ZoS10jKDRb6gc8USDXzmi+AUzyxdOzqgutQu5V7cojWD12cTthbmmMs9TcStgQRNTbxcppUF6atvuOhunKQOu6ndHKnp5G//KjNKnDNVWUIGqLWV9pqQgm37No1cfXU1r+w9evCAgBlKbOlQ0Ro7FsRp30EcH/Fe5HVrWS0WH3M4EV6+8IL4EZpNz5GiYs2FRF89bbxFUr/UGNDa0ZpaFt1DFS30uW0iVD7ONvURAU5EwPjo+RTO30Ov7zzPkeQg69SeN8Lw7a7t9/RnseoeKYmJpZEeu0j3eBEtDyTfdl1'
```

**Comment by LRagji** [comment]
Seems correct

**Comment by girishg4t** [comment]
@LRagji  i am thinking of storing this data in github same as we do for EQ https://github.com/girishg4t/nse-bse-bhavcopy
for that i want to understand how the option data changes i.e. if i downloaded the data today how long i can use that ?
do we need to downolad it every week, day, month ? i think it is week i.e. every thursday or friday morning ?

 could you please guide me on this, i don't have much knowledge on options

i want to download the data and store that in csv or any other format for a week(or utill it get change).

**Comment by LRagji** [comment]
1. Perfect, please store the data in the same repo but under different folder "derivatives" just giving a name, I would further like to partition the data according to weeks into different folders for easy searching.
2. This should be daily data downloaded by EOD, what you are downloading is called option chain, Each option(Instruments named as Call & Put) has an weekly expiry for a given strike price., They change every time as market moves, but this daily file should be good enough..

**Comment by girishg4t** [comment]
Hey @LRagji ,
I have now added the options data, it is currently in json format, we can format it later into csv
Below is the location
https://github.com/girishg4t/nse-bse-bhavcopy/tree/master/OPTIONS/NSE
